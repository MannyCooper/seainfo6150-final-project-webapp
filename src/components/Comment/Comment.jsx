import React, { useState } from "react";
import styles from "./Comment.module.css"

let key = 0;
const Comment = () => {
  const [submittedForm, setSubmittedForm] = useState([]);
  function onSubmit(e) {
    e.preventDefault();
    if (validateRequiredForm()) {
      const comment = new FormData(e.target);
      if (validateEmail(comment.get("email"))) {
        if (validateAge(comment.get("age"))) {
          console.log("Response", comment.get("response"));
          setSubmittedForm(submittedForm.concat(result(comment)));
          document.getElementById("form").reset()
        }
      }
    }
  };

  function validateRequiredForm() {
    let name = document.forms["comment"]["name"].value;
    let email = document.forms["comment"]["email"].value;
    let content = document.forms["comment"]["content"].value;

    if (name == null || name === "") {
      alert("Name is required!");
      return false;
    } else {
      if (email == null || email === "") {
        alert("Email is required!");
        return false;
      } else {
        if (content == null || content === "") {
          alert("Content is required!");
          return false;
        } else {
          return true;
        }
      }
    }
  }

  function validateEmail(email) {
    let re = /\S+@\S+\.\S+/;
    if (re.test(email)) {
      return true;
    } else {
      alert("Wrong Email Format!");
      return false;
    };
  }

  function validateAge(age) {
    let re = /^[1-9][0-9]?$|^100$/;
    if (re.test(age) || age == null || age === "") {
      return true;
    } else {
      alert("Age should be between 1 and 100!");
      return false;
    };
  }


  function result(form) {
    console.log(form);
    let result = (
      <div key={key++}>
        <p>Name: {form.get("name")}, Email: {form.get("email")}, Country: {form.get("country")}</p>
        <p>Age: {form.get("age")}, Gender: {form.get("genderDropdown")}, Job: {form.get("jobDropdown")}</p>
        <p>Response: {form.get("response")}</p>
        <p>Content: {form.get("content")}</p>
      </div>
    )
    return result;
  }

  return (
    <div className={styles.comment_container}>
      <h2 className={styles.title}>Comment</h2>
      {
        (
          <div>
          <form id="form" className={styles.form} name="comment" onSubmit={onSubmit}>

            <div className={styles.text}>
              <label htmlFor="name">Name:</label>
              <input type="text" name="name" id="name" placeholder="Required" />
            </div>

            <div className={styles.text}>
              <label htmlFor="email">Email:</label>
              <input type="email" name="email" id="email" placeholder="Required and Validated"></input>
            </div>

            <div className={styles.text}>
              <label htmlFor="country">Country:</label>
              <input type="text" name="country" id="country"></input>
            </div>

            <div className={styles.text}>
              <label htmlFor="age">Age:</label>
              <input type="text" name="age" id="age" placeholder="Validated (1-100)"></input>
            </div>

            <div className={styles.dropdown}>
              <label htmlFor="gender">Gender:</label>
              <select name="genderDropdown" id="gender">
                <option value="male">Male</option>
                <option value="female">Female</option>
                <option value="other">Other</option>
              </select>
            </div>

            <div className={styles.dropdown}>
              <label htmlFor="job">Job:</label>
              <select name="jobDropdown" id="job">
                <option value="programmer">Programmer</option>
                <option value="designer">Designer</option>
                <option value="other">Other</option>
              </select>
            </div>

            <div className={styles.radio}>
              <label>Response:</label>
              <label htmlFor="responseGood"><span role="img" aria-label="Good">üëç</span></label>
              <input type="radio" name="response" id="responseGood" value="üëç" />
              <label htmlFor="responseFair"><span role="img" aria-label="Fair">ü§î</span></label>
              <input type="radio" name="response" id="responseFair" value="ü§î" />
              <label htmlFor="responseBad"><span role="img" aria-label="Bad">üëé</span></label>
              <input type="radio" name="response" id="responseBad" value="üëé" />
            </div>

            <div className={styles.comment_content}>
              <label htmlFor="commentContent">Content</label>
              <textarea name="content" rows="4" cols="50" id="commentContent" placeholder="Required" />
            </div>
            <input className={styles.button} type="submit" value="Send" />
          </form>
          </div>
        )
      }
      {(
        submittedForm ? submittedForm : null)
      }
    </div>
  )
}

export default Comment
