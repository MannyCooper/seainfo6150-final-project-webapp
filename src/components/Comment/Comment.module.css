@import url('https://fonts.googleapis.com/css2?family=Lato&display=swap');

.comment_container {
    max-width: 1100px;
    margin: 30px auto;
    border-radius: 25px;
    padding: 30px;
    box-shadow: 0 4px 10px rgba(0, 0, 0, .05), 0 0 1px rgba(0, 0, 0, .1);
    color: #4a4a4a;
}

.comment_container label {
    font-family: 'Lato', sans-serif;
}

.title {
    font-size: 32px;
    line-height: 1.1;
    font-weight: 700;
    letter-spacing: 0.004em;
    margin-bottom: 24px;
}

.form {
    display: grid;
    grid-template-columns: 1fr 1fr 1fr 1fr;
    grid-template-rows: 1fr 1fr 1fr;
    gap: 0px 0px;
}

.radio {
    grid-area: 2 / 3 / 3 / 5;
}

.comment_content {
    grid-area: 3 / 1 / 4 / 3;
}

.button {
    width: 100px;
    height: 50px;
}


.form label {
    display: block;
}

.radio label:nth-child(n+2) {
    display: inline;
}

@media only screen and (max-width: 1380px) {
    .comment_container {
        max-width: 980px;
    }
}

@media only screen and (max-width: 1200px) {
    .comment_container {
        width: 692px;
    }
    .form textarea{
        max-width: 300px;
    }
}

@media only screen and (max-width: 768px) {
    .comment_container {
        max-width: 366px;
        width: 87.5%;
    }

    .form {
        grid-template-columns: 1fr 1fr;
        grid-template-rows: 1fr 1fr 1fr 1fr;
        grid-template-areas:
            ". ."
            ". ."
            ". ."
            "radio comment_content";
    }

    .radio {
        grid-area: radio;
    }

    .comment_content {
        grid-area: comment_content;
    }
    .form input {
        max-width: 120px;
    }
    .form textarea{
        max-width: 150px;
    }
}