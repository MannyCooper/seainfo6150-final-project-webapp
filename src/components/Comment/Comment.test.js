 /* global render */
 import React from "react";
 import Comment from "./Comment.jsx";
  
 it("renders the Comment component correctly", () => {
   const { container } = render(<Comment />);
   expect(container).toMatchSnapshot();
 });
 