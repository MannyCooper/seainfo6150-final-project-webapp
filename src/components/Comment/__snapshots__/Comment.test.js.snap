// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders the Comment component correctly 1`] = `
<div>
  <div
    class="comment_container"
  >
    <h2
      class="title"
    >
      Comment
    </h2>
    <div>
      <form
        class="form"
        id="form"
        name="comment"
      >
        <div
          class="text"
        >
          <label
            for="name"
          >
            Name:
          </label>
          <input
            id="name"
            name="name"
            placeholder="Required"
            type="text"
          />
        </div>
        <div
          class="text"
        >
          <label
            for="email"
          >
            Email:
          </label>
          <input
            id="email"
            name="email"
            placeholder="Required and Validated"
            type="email"
          />
        </div>
        <div
          class="text"
        >
          <label
            for="country"
          >
            Country:
          </label>
          <input
            id="country"
            name="country"
            type="text"
          />
        </div>
        <div
          class="text"
        >
          <label
            for="age"
          >
            Age:
          </label>
          <input
            id="age"
            name="age"
            placeholder="Validated (1-100)"
            type="text"
          />
        </div>
        <div
          class="dropdown"
        >
          <label
            for="gender"
          >
            Gender:
          </label>
          <select
            id="gender"
            name="genderDropdown"
          >
            <option
              value="male"
            >
              Male
            </option>
            <option
              value="female"
            >
              Female
            </option>
            <option
              value="other"
            >
              Other
            </option>
          </select>
        </div>
        <div
          class="dropdown"
        >
          <label
            for="job"
          >
            Job:
          </label>
          <select
            id="job"
            name="jobDropdown"
          >
            <option
              value="programmer"
            >
              Programmer
            </option>
            <option
              value="designer"
            >
              Designer
            </option>
            <option
              value="other"
            >
              Other
            </option>
          </select>
        </div>
        <div
          class="radio"
        >
          <label>
            Response:
          </label>
          <label
            for="responseGood"
          >
            <span
              aria-label="Good"
              role="img"
            >
              üëç
            </span>
          </label>
          <input
            id="responseGood"
            name="response"
            type="radio"
            value="üëç"
          />
          <label
            for="responseFair"
          >
            <span
              aria-label="Fair"
              role="img"
            >
              ü§î
            </span>
          </label>
          <input
            id="responseFair"
            name="response"
            type="radio"
            value="ü§î"
          />
          <label
            for="responseBad"
          >
            <span
              aria-label="Bad"
              role="img"
            >
              üëé
            </span>
          </label>
          <input
            id="responseBad"
            name="response"
            type="radio"
            value="üëé"
          />
        </div>
        <div
          class="comment_content"
        >
          <label
            for="commentContent"
          >
            Content
          </label>
          <textarea
            cols="50"
            id="commentContent"
            name="content"
            placeholder="Required"
            rows="4"
          />
        </div>
        <input
          class="button"
          type="submit"
          value="Send"
        />
      </form>
    </div>
  </div>
</div>
`;
